### PR TITLE
Implement get full session

### PR DESCRIPTION
**Pull request by glsubri**
This will fix the second point of issue #124 

Note that the server still doesn't respond with the correct json session representation:
```
{
  'idSession': {'idSession': 1},
  'timestampStart': '2020-04-19T07:07:37.952+0000',
  'timestampEnd': None,
  'code': '0xb5c4',
  'state': 'OPEN'
}
```
instead of 
```
{
  'idModerator': 1,
  'idPoll': 1,
  'idSession': 1,
  'timestampStart': '2020-04-19T07:07:37.952+0000',
  'timestampEnd': None,
  'code': '0xb5c4',
  'state': 'OPEN'
}
```

### CODE REVIEW

File: src/main/java/ch/heigvd/pro/b04/polls/exceptions/IllegalPollStateException.java [hunk 1]
```diff
@@ -0,0 +1,7 @@
+package ch.heigvd.pro.b04.polls.exceptions;
+
+public class IllegalPollStateException extends IllegalStateException {
```
Comment (alexandrepiveteau) [inline src/main/java/ch/heigvd/pro/b04/polls/exceptions/IllegalPollStateException.java]: IllegalPollStateException is not checked, but we probably want to make it checked to enforce `try ... catch ...` blocks around methods that might throw it.

File: src/main/java/ch/heigvd/pro/b04/moderators/Moderator.java [hunk 2]
```diff
@@ -89,4 +94,47 @@ public ServerPoll newPoll(ServerPollRepository repository, ClientPoll poll) {
...
+   *         doesn't exist
+   */
+  public ServerPoll getPollWithId(ServerPollRepository serverPollRepository, Integer idPoll)
+      throws PollNotExistingException {
+
+    List<ServerPoll> pollList = serverPollRepository.findByModeratorAndId(this, idPoll);
+    if (pollList.isEmpty()) {
+      throw new PollNotExistingException();
+    } else if (pollList.size() > 1) {
+      throw new IllegalPollStateException();
```
Comment (alexandrepiveteau) [inline src/main/java/ch/heigvd/pro/b04/moderators/Moderator.java]: Unchecked `IllegalPollStateException` (more info on the comment of the class itself).

File: src/main/java/ch/heigvd/pro/b04/moderators/Moderator.java [hunk 3]
```diff
@@ -89,4 +94,47 @@ public ServerPoll newPoll(ServerPollRepository repository, ClientPoll poll) {
...
+   * @param serverPollRepository The serverpoll repository
+   * @param idPoll The id of the poll
+   * @return The poll associated with idPoll if it belongs to the moderator
+   * @throws PollNotExistingException is thrown if the poll doesn't belong to the moderator or
+   *         doesn't exist
+   */
+  public ServerPoll getPollWithId(ServerPollRepository serverPollRepository, Integer idPoll)
+      throws PollNotExistingException {
+
+    List<ServerPoll> pollList = serverPollRepository.findByModeratorAndId(this, idPoll);
```
Comment (alexandrepiveteau) [inline src/main/java/ch/heigvd/pro/b04/moderators/Moderator.java]: Spring may actually be able to return an `Optional<ServerPoll>`, which would better define the semantics of the method.

File: src/main/java/ch/heigvd/pro/b04/sessions/SessionController.java [hunk 4]
```diff
@@ -188,4 +178,37 @@ public ServerSession putSession(
...
+   * @param token The token of the moderator
+   * @return The last ServerSession
+   * @throws WrongCredentialsException is thrown if the moderator and the token don't match or if
+   *         the moderator doesn't exist
+   * @throws PollNotExistingException is thrown if the poll doesn't belong to the moderator or if
+   *         the poll doesn't exist
+   * @throws SessionNotExistingException is thrown if the poll never had a session
+   */
+  @GetMapping(value = "/mod/{idModerator}/poll/{idPoll}/session")
+  public ServerSession getLastActiveSession(
```
Comment (alexandrepiveteau) [inline src/main/java/ch/heigvd/pro/b04/sessions/SessionController.java:196]: `@Transactional`.

**Review by alexandrepiveteau — CHANGES_REQUESTED** [review]

File: src/main/java/ch/heigvd/pro/b04/moderators/Moderator.java (hunk 2 above)
Comment (glsubri) [reply src/main/java/ch/heigvd/pro/b04/moderators/Moderator.java]: This Exception is deleted by the refactoring of `findByModeratorAndId`

**Review by alexandrepiveteau — APPROVED** [review]

**Pull request merged**